src/coreclr/src/jit/gentree.h

```diff
@@ -3108,12 +3124,193 @@ struct GenTreeLclVarCommon : public GenTreeUnOp
+// specified by its index.
+//
+// Arguments:
+//    idx  -  Position or index of the return register
+//
+// Return Value:
+//    Returns GTF_* flags associated with the register. Only GTF_SPILL and GTF_SPILLED are considered.
+//
+inline unsigned GetMultiRegSpillFlagsByIdx(MultiRegSpillFlags flags, unsigned idx)
+{
+    static_assert_no_msg(MAX_RET_REG_COUNT * 2 <= sizeof(unsigned char) * BITS_PER_BYTE);
+    assert(idx < MAX_RET_REG_COUNT);
+
+    unsigned bits       = flags >> (idx * 2); // It doesn't matter that we possibly leave other high bits here.
+    unsigned spillFlags = 0;
+    if (bits & PACKED_GTF_SPILL)
+    {
+        spillFlags |= GTF_SPILL;
+    }
+    if (bits & PACKED_GTF_SPILLED)
+    {
+        spillFlags |= GTF_SPILLED;
+    }
+    return spillFlags;
+}
+
+//----------------------------------------------------------------------
+// SetMultiRegSpillFlagsByIdx: set spill flags for the register specified by its index.
+//
+// Arguments:
+//    oldFlags   - The current value of the MultiRegSpillFlags for a node.
+//    flagsToSet - GTF_* flags. Only GTF_SPILL and GTF_SPILLED are allowed.
+//                 Note that these are the flags used on non-multireg nodes,
+//                 and this method adds the appropriate flags to the
+//                 incoming MultiRegSpillFlags and returns it.
+//    idx    -     Position or index of the register
+//
+// Return Value:
+//    The new value for the node's MultiRegSpillFlags.
+//
+inline MultiRegSpillFlags SetMultiRegSpillFlagsByIdx(MultiRegSpillFlags oldFlags, unsigned flagsToSet, unsigned idx)
+{
+    static_assert_no_msg(MAX_RET_REG_COUNT * 2 <= sizeof(unsigned char) * BITS_PER_BYTE);
+    assert(idx < MAX_RET_REG_COUNT);
+
+    MultiRegSpillFlags newFlags = oldFlags;
+    unsigned           bits     = 0;
+    if (newFlags & GTF_SPILL)
+    {
+        bits |= PACKED_GTF_SPILL;
+    }
+    if (newFlags & GTF_SPILLED)
+    {
+        bits |= PACKED_GTF_SPILLED;
+    }
+
+    const unsigned char packedFlags = PACKED_GTF_SPILL | PACKED_GTF_SPILLED;
+
+    // Clear anything that was already there by masking out the bits before 'or'ing in what we want there.
+    newFlags = (unsigned char)((newFlags & ~(packedFlags << (idx * 2))) | (bits << (idx * 2)));
+    return newFlags;
+}
+
 // gtLclVar -- load/store/addr of local variable
 
 struct GenTreeLclVar : public GenTreeLclVarCommon
 {
+private:
+    regNumberSmall     gtOtherReg[MAX_MULTIREG_COUNT - 1];
+    MultiRegSpillFlags gtSpillFlags;
+    unsigned int GetLastUseBit(int regIndex)
+    {
+        assert(regIndex < 4);
+        return (regIndex == 0) ? GTF_VAR_DEATH : (1 << (MULTIREG_LAST_USE_SHIFT + regIndex));
+    }
+
+public:
     INDEBUG(IL_OFFSET gtLclILoffs;) // instr offset of ref (only for JIT dumps)
 
+    // Multireg support
+    bool IsMultiReg() const
+    {
+        return ((gtFlags & GTF_VAR_MULTIREG) != 0);
+    }
+    void ClearMultiReg()
```

Comment:
In which situations could we clear that flag?

Reply:
The flag means that the node represents a multireg localVar, and it is only set when all the fields are register candidates. If that changes, the flag is cleared. This makes it a bit easier to determine the right code path to take without extra checking.